src/App.tsx

```diff
@@ -15,48 +15,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NAME = 'name',
+  LENGTH = 'length',
+}
+
+function sortArray(
+  goods: string[],
+  howSort: SortType,
+  isReversed: boolean,
+) {
+  const array = [...goods];
+
+  array.sort((value1, value2) => {
+    if (howSort === SortType.NAME) {
+      return value1.localeCompare(value2);
+    }
+
+    if (howSort === SortType.LENGTH) {
+      return value1.length - value2.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    array.reverse();
+  }
+
+  return array;
+}
+
 export const App: React.FC = () => {
+  const [sort, setSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortArray(
+    goodsFromServer,
+    sort as SortType,
+    isReversed,
+  );
+
+  function resetAll() {
+    setSort('');
+
+    setIsReversed(false);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sort === SortType.NAME ? null : 'is-light'}`}
+          onClick={() => setSort(SortType.NAME)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${sort === SortType.LENGTH ? null : 'is-light'}`}
+          onClick={() => setSort(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${isReversed === true ? null : 'is-light'}`}
+          onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {sort !== '' || isReversed
+          ? (
+            <button
+              type="button"
+              className={`button is-danger ${sort !== '' || isReversed ? null : 'is-light'}`}
+              onClick={resetAll}
+            >
+              Reset
+            </button>
+          )
+          : null}
```

Comment:
Seems that the ternary is redundant here. Use `&&` instead

Reply:
Still not fixed